meta-refkit-extra/conf/distro/include/refkit-extra.conf

```diff
@@ -4,3 +4,6 @@ SUPPORTED_RECIPES_append = " \
      ${META_REFKIT_EXTRA_BASE}/conf/distro/include/refkit-extra-supported-recipes.txt \
 "
 SUPPORTED_RECIPES[refkit-extra-supported-recipes.txt] = "extra"
+
+# Caffe needs Boost Python 3 bindings. Enable them in Boost.
+PACKAGECONFIG_pn-boost += "python"
```

Comment:
I think you may need PACKAGECONFIG_append_pn-boost = " python" here. 

PACKAGECONFIG_pn-boost overrides the weaker PACKAGECONFIG default instead of extending it.